laplace/utils/matrix.py

```diff
@@ -72,7 +68,7 @@ def __add__(self, other):
         if not isinstance(other, Kron):
             raise ValueError('Can only add Kron to Kron.')
 
-        kfacs = [[Hi.add(Hj) for Hi, Hj in zip(Fi, Fj)]
+        kfacs = [[Hi + Hj for Hi, Hj in zip(Fi, Fj)]
```

Comment:
Why this change? I remember there was a reason for using `.add`, might have had to do with differentiability.

Reply:
I think this is due to changing the initialization of the Kronecker factors to not assume a specific shape by initializing with `0.0` scalars, i.e. not a tensor so `.add` would not work anymore. The motivation for this was to support diagonal Kronecker factors for large layers like embedding layers in language models. See the commit [here](https://github.com/aleximmer/Laplace/pull/144/commits/98a9800db0a28c023ba73444be3cd8d490eeca62).

Reply:
Not a big fan of this design, so not sure if it should be merged into `main`. Also, regarding @aleximmer's comment, are there tests for checking if the differentiability is affected?

Reply:
I take the point of not assuming a specific shape. I think in the future this is still necessary, but let's build it from the ground up for the next release (0.3). Perhaps along with the addition of `SINGD` backend.

For the 0.2 milestone, let's revert this. (This change also breaks https://github.com/aleximmer/Laplace/issues/171 BTW)